### PR TITLE
Workaround for missing debug info

### PR DESCRIPTION
**Pull request by tmat**
Do not emit hidden sequence point on closure allocation when the closure scope is switch expression.

### CODE REVIEW

**Comment by tmat** [comment]
@dotnet/roslyn-compiler PTAL. Test only.

File: src/Compilers/CSharp/Test/Emit/PDB/PDBTests.cs [hunk 1]
```diff
@@ -7318,6 +7318,42 @@ public void SyntaxOffset_Pattern()
 </symbols>");
         }
 
+        [ConditionalFact(typeof(WindowsOnly), Reason = ConditionalSkipReason.NativePdbRequiresDesktop)]
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/PDB/PDBTests.cs]: It's possible that we fixed the PDB testing infrastructure to support Core too. Try with `[Fact]`?

Comment (tmat) [reply src/Compilers/CSharp/Test/Emit/PDB/PDBTests.cs]: I'll follow up and flip all tests.

**Comment by tmat** [comment]
Update: I have implemented a workaround that should address the complete lack of debug info. Leaving the issue open as the workaround is not an ideal solution.

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler This PR from Tomas on emitting switch expressions could use a second review/sign-off. Thanks

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler This PR from Tomas on emitting switch expressions could use a second review/sign-off. Thanks

**Comment by tmat** [comment]
@dotnet/roslyn-compiler I'd actually prefer a real fix to be implemented instead of this workaround.

**Comment by jcouv** [comment]
> I'd actually prefer a real fix to be implemented instead of this workaround.

I take it that this PR should not be merged as-is. I'll mark as private for now.

**Pull request closed without merging**